crates/solver/src/driver_logger.rs

```diff
@@ -147,13 +147,24 @@ impl DriverLogger {
         let simulation_gas_limit = self.simulation_gas_limit;
         let task = async move {
             let simulations = simulate_and_error_with_tenderly_link(
-                errors.iter().map(|(solver, settlement, access_list, _)| {
-                    (
-                        solver.account().clone(),
-                        settlement.clone(),
-                        access_list.clone(),
-                    )
-                }),
+                errors.iter().map(
+                    |SimulationWithError {
+                         simulation:
+                             Simulation {
+                                 solver,
+                                 settlement,
+                                 transaction,
+                                 ..
+                             },
+                         ..
+                     }| {
+                        (
+                            solver.account().clone(),
+                            settlement.clone(),
+                            transaction.access_list.clone(),
+                        )
```

Comment:
style nit: The destructuring looks quite noisy here. What do you think about something like 👇  instead?
The same idea could also be applied to the destructuring in the for loop.

```suggestion
                    |simulation_with_error| {
                        let simulation = &simulation_with_error.simulation;
                        (
                            simulation.solver.account().clone(),
                            simulation.settlement.clone(),
                            simulation.transaction.access_list.clone(),
                        )
                    }
```

Reply:
Yes, I've abused a bit the pattern matching here :) Fixed.